src/test/ui/consts/std/alloc.32bit.stderr

```diff
@@ -1,14 +1,25 @@
 error[E0080]: it is undefined behavior to use this value
-  --> $DIR/alloc.rs:8:1
+  --> $DIR/alloc.rs:9:1
    |
-LL | const LAYOUT_INVALID: Layout = unsafe { Layout::from_size_align_unchecked(0x1000, 0x00) };
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed at .align_: encountered 0, but expected something greater or equal to 1
+LL | const LAYOUT_INVALID_ZERO: Layout = unsafe { Layout::from_size_align_unchecked(0x1000, 0x00) };
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type validation failed at .align_.0.<enum-tag>: encountered 0x00000000, but expected a valid enum tag
```

Comment:
This error message is not exactly clear.

Reply:
I think that's ok.  It's a best-effort UB check -- the place to look is at the `# Safety` comment, not the error message.

One gets essentially the same error from things like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c2b19c953a711335567cc3fa7c6ad4f9) too:
```rust
pub const BAD_ORDERING: std::cmp::Ordering = unsafe { std::mem::transmute(3_u8) };
```

So it'll get better here if the error in general gets better for enums, which seems fine for this PR.